services/src/handlers/projects.rs

```diff
@@ -669,6 +671,183 @@ mod tests {
+            let res = warp::test::request()
+                .method("PATCH")
+                .path(&format!("/project/{}", project_id.to_string()))
+                .header("Content-Length", "0")
+                .header(
+                    "Authorization",
+                    format!("Bearer {}", session.id.to_string()),
+                )
+                .json(&update)
+                .reply(&update_project_handler(ctx.clone()))
+                .await;
+
+            assert_eq!(res.status(), 200);
+
+            let loaded = ctx
+                .project_db()
+                .read()
+                .await
+                .load_latest(session.user.id, project_id)
+                .await
+                .unwrap();
+
+            loaded.layers
+        }
+
+        let ctx = InMemoryContext::default();
+
+        ctx.user_db()
+            .write()
+            .await
+            .register(
+                UserRegistration {
+                    email: "foo@bar.de".to_string(),
+                    password: "secret123".to_string(),
+                    real_name: "Foo Bar".to_string(),
+                }
+                .validated()
+                .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        let session = ctx
+            .user_db()
+            .write()
+            .await
+            .login(UserCredentials {
+                email: "foo@bar.de".to_string(),
+                password: "secret123".to_string(),
+            })
+            .await
+            .unwrap();
+
+        let project = ctx
+            .project_db()
+            .write()
+            .await
+            .create(
+                session.user.id,
+                CreateProject {
+                    name: "Test".to_string(),
+                    description: "Foo".to_string(),
+                    bounds: STRectangle::new(
+                        SpatialReferenceOption::Unreferenced,
+                        0.,
+                        0.,
+                        1.,
+                        1.,
+                        0,
+                        1,
+                    )
+                    .unwrap(),
+                }
+                .validated()
+                .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        let layer_1 = Layer {
+            workflow: WorkflowId::new(),
+            name: "L1".to_string(),
+            info: LayerInfo::Raster(RasterInfo {
+                colorizer: Colorizer::Rgba,
+            }),
+        };
+
+        let layer_2 = Layer {
+            workflow: WorkflowId::new(),
+            name: "L2".to_string(),
+            info: LayerInfo::Vector(VectorInfo {}),
+        };
+
+        // add first layer
+        assert_eq!(
+            update_and_load_latest(
+                &ctx,
+                &session,
+                project,
+                UpdateProject {
+                    id: project,
+                    name: None,
```

Comment:
another problem with this approach of handling updates is: what about optional fields? How do we distinguish between "No Update" and "Set value to None". We need a different approach here. My suggestion would be to directly adress the fiels in the URI so "PATCH /project/uuid/name" and set a value. But for now we can do it like this if you want.

Reply:
I did not change the way layer updates work, but layer list updates.

Reply:
is there a difference?

Reply:
Ah, I misread the code location. I did not change the `UpdateLayer` mechanism, just the layer portion. I guess optional fields are a separate task. As for layers, maybe a binary option is not enough.